src/Stubs/BaseStub.php

```diff
@@ -82,6 +82,16 @@ public function getCalls(string $method): array
         return $this->calls[$method] ?? [];
     }
 
+    /**
+     * Resets the stub's record of all calls.
+     *
+     * @return void
+     */
+    public function resetStub(): void
```

Comment:
shouldn't stubs be single use?

Reply:
Its a good question - we're using the SearchClientStub inside functional tests where we need to be able to reset them into an empty state after the arrange phase of a test (setting up precondition entities, we only want to assert against changes made during the act phase)

Reply:
I moved the function from SearchClientStub to Stub thinking it'd be helpful in a general case but I'm happy to move it back

Reply:
I think it's ok, because it's not the generic stub used for everything.

Does it need to clear `$responses` also?

Reply:
Potentially, but that opens the can of worms about should the stub be reconfigurable and set new responses or not?

Maybe rename the method to resetStubCalls() to make it clear it is only resetting the recorded calls?

Reply:
True. I think that can of worms answers if it should go back into the SearchClientStub then.

Reply:
$calls is private, decided to rename to resetStubCalls() to make it clear only the calls are reset.